src/components/carousel/index.jsx

```diff
@@ -0,0 +1,70 @@
+import PropTypes from "prop-types";
+import Slider from "react-slick";
+import { useEffect, useRef } from "react";
+
+const COMPONENT_CLASS_NAME = "c-carousel";
+
+const Carousel = ({ children, className, sliderProps }) => {
+	const sliderRef = useRef();
+	const { onReInit: sliderReInit } = sliderProps;
+
+	/**
+	 * This useEffect will enhance a11y in the following:
+	 * 1) It will add the role and aria-label to the react-slick component
+	 * 2) It will add needed roles and labels to each slide.
+	 * 3) If paging is being used it will set the current pager container to aria-current=true
+	 */
+	useEffect(() => {
+		const slider = sliderRef.current.querySelector(".slick-slider");
+		slider?.setAttribute("role", "region");
+		slider?.setAttribute("aria-label", "carousel");
+
+		const slides = slider?.querySelectorAll(".slick-slide");
+		slides?.forEach((slide) => {
+			slide.setAttribute("role", "group");
+			slide.setAttribute("aria-label", `slide ${parseInt(slide.dataset.index, 10) + 1}`);
```

Comment:
It would have to be an array of values or an attribute on the child elements of the carousel, unless you are thinking that "slide" would be translated.  If we wanted to allow them to label each slide individually we'd have to require that the user passes an array of labels or they add a `data-label` (or a `title`) that we can use to label the slide.